### ISSUE TITLE
Pageless content section by default on initial install?

### ISSUE THREAD

**Issue by philwareham** [textpattern/textpattern#1572]
### Is your feature request related to a problem?

No. A suggestion for debate.

### What is the feature?

We now allow [pageless sections](https://textpattern.com/weblog/426/feature-focus-live-pageless-sections-for-hidden-content) to silo content that won't be part of the standard article flow that Textpattern traditionally had for years. That's great, now I've worked out how to use it.

I'm wondering if there is any merit in providing a section for this when Textpattern is first installed (as we already do for `articles`). Maybe called `pageless-content` or something similar?

The primary reason behind this is related to themes: I am building a theme that extensively uses a concept of 'modules' you can bolt together quickly to build a site. As part of that I want certain modules to contain content - but that content to **not** be part of the standard article flow.

My thinking is that, for example, if you wanted a snappy intro on your homepage you could choose a text layout from the various text modules I'll provide in the theme, and then call the text from the `pageless-content` section maybe like so:

    <txp:article form="-module--text-layout--1" id="2" />

Where the article with id2 is stored in the `pageless-content` section.

I know I could (and will) provide instructions in the theme to outline how to create this section and populate it - by having it in a default install it takes that barrier away slightly, and also regardless of theming it gives a hint to new users that this feature exists.

Obviously the section could be deleted by users if they don't want it - I'm not suggesting it's a hard-coded in like the `default` section is.

Not sure whether to also suggest a second article in the default install that is housed in the pageless section, and provides a very brief overview of what it is? That might be a step too far, but could be very handy.

Thoughts?

**Comment by petecooper** [comment]
Interesting idea. Respectfully, a light -1 from me.

>That's great, now I've worked out how to use it.

That's the pull quote. I haven't got a clue how they work, not used them, probably will at some stage when I find a use.

For a default installation, the barrier to entry needs to be _low_. I think there's too much text in article ID 1 as it stands, and it could be pared back. The fundamentals should be in the default stuff, and pageless sections are beyond the fundamentals in my view. Perfect for a 'next steps' mini site, but considering the terminology that people need to get their head around (and we're already a bit niche in that respect), it's a bit too much too soon.

**Comment by philwareham** [comment]
Thanks Pete, valid points as always. I'm not so sure though.

I fully agree that the welcome article is way too wordy - maybe we should discuss that as a separate issue (especially as I do want to complete the [localised welcome text issue](https://github.com/textpattern/textpattern/issues/1179) sometime relatively soon).

However, pageless content is a fundamental new feature to both present and future workflow IMO (especially so once unlimited custom fields land) as it creates silos of user-editable content outside of the normal (enforced) site structure.

Previously my way round this was to simply hard-code content such as this within a page/form template - but that is not overly satisfactory since:

1. It mixes content with structure.
2. It makes it difficult for end-users (authors) to amend said content.
3. It makes it nigh on impossible to create modular themes.

By implicitly showing content 'silos' are possible makes users aware that they can be used as part of your site build process.

**Comment by philwareham** [comment]
Basically, if/when unlimited custom fields land, the above mentioned feature could bring Craft CMS levels of content management to Textpattern. i.e. user-defined snippets on content for use anywhere on your site (and even off-site, given the right API).

**Comment by petecooper** [comment]
>I fully agree that the welcome article is way too wordy - maybe we should discuss that as a separate issue

+1 from me.

On the pageless sections, I agree 100% that it's a fundamental new feature, totally. We're also a decade-plus ahead of most people trying/using Textpattern for the first time, and we know about the finer points of pages, forms, sections, categories, styles, and so on.

Excluding themes and plugins for now as they're fairly commonly understood across the CMS landscape, the other terminology is still open for interpretation to newbies. They need to know what a page and a section is before they can marry the two together, and understand that a page can be assigned to a section…or is the section assigned to a page?

But the section doesn't always have to use a page…apart from when it does…and this is where we lose them with terminology technobabble or logic overload. Someone who just wants to write isn't going to give a care about it, and devs who _do_ want to tinker will benefit from a nice docs page with explanations and copious examples to try.

It's a _great_ CMS, we're just not very good at selling it at the moment.

>if/when unlimited custom fields land, the above mentioned feature could help bring Craft CMS levels of content management to Textpattern

Yep, definitely. I suspect we're at least a year or two away from that, realistically.

**Comment by petecooper** [comment]
>I fully agree that the welcome article is way too wordy - maybe we should discuss that as a separate issue

Done: https://github.com/textpattern/textpattern/issues/1574

**Comment by philwareham** [comment]
> They need to know what a page and a section is before they can marry the two together

Yeah agreed there, I can understand _why_ @bloke called it 'pageless content' (because the content has been separated from needing a page template to display it), but it could also be 'sectionless content' (although technically it _has_ a section, just not viewable). Both those terms only make sense if you understand Textpattern's standard workflow already though.

**Comment by petecooper** [comment]
>Both those terms only make sense if you understand Textpattern's standard workflow already though.

^ This. You need to know how to order a beer in Textpatternese before you can start brewing your own, or some similar awkward analogy.

**Comment by philwareham** [comment]
True. Going back to my first post, this is primarily to make using themes a bit easier for both end-users and for theme authors. I don't really want to have to explain the inner workings of how to use Textpattern just to get a theme up and running. It needs to be made as possible using the tools we already have at our disposal.

Remember that many potential users come and look at Textpattern, realise no themes are easily installable (yet) and then leave. I'm just trying to lower that barrier to entry a little.

Maybe partnered with a shorter welcome article - a small pageless content article addition would be agreeable. Or maybe not.

**Comment by petecooper** [comment]
>Remember that many potential users come and look at Textpattern, realise no themes are easily installable (yet) and then leave.

Yep, that's the more fundamental problem that needs addressing – but it's a chicken and egg situation at the moment with theme availability. When there's ~10 or so _solid_ themes that can be deployed easily enough, a bullet point "how to" list or even a screencast showing how it's done (i.e. visual people often like visual methods, nerds often prefer text), _that's_ the winning formula and it'll be way easier.

Anyone serious about making a site with a pre-made theme on an empty Textpattern would be better off just installing the blank theme on install, getting the pretty theme installed, and then building out any page/section scaffold according to the theme author's recipe.

**Comment by bloatware** [comment]
I see the point of the OP, but still am for making the default theme as hello-worldish (but nice) as reasonably possible. It should be easy to grasp for newcomers.

**Comment by philwareham** [comment]
Yes, this isn't for the default theme, specifically (I don't have any plans to change that in any major way). Rather, for our own, more advanced themes.

I can see I'm in the minority here, so I guess I'll just have to document the process as best I can with the theme and hope users can grasp it. Will close.
